src/magnetometer.rs

```diff
@@ -54,6 +54,25 @@ where
             z: data.2 as i16,
         })
     }
+
+    /// Enable or disable the magnetometer's built in offset cancellation
+    ///
+    /// Offset cancellation is **automatically** managed by the device in **continuous** mode.
+    ///
+    /// Offset cancellation has to be **managed by the user** in **single measurement** (OneShot) mode averaging
+    /// two consecutive measurements H<sub>n</sub> and H<sub>n-1</sub>.
+    pub fn mag_offset_cancellation(&mut self, enable: bool) -> Result<(), Error<CommE, PinE>> {
+        let cfg = self.cfg_reg_b_m.bits & 0b11111101;
+
+        let mask = if enable { 1 << 1 } else { 0 };
+
+        self.iface
+            .write_mag_register(Register::CFG_REG_B_M, cfg | mask)?;
+
+        self.cfg_reg_b_m = (cfg | mask).into();
```

Comment:
Could you do these bit flag manipulations by defining variants in the [`BitFlags` `struct`](https://github.com/eldruin/lsm303agr-rs/blob/ba698f7cf1830fc5a6a7d4064d8bb97efdfa729c/src/register_address.rs#L22) and then using the methods `with_high` and `with_low`. See for example [this](https://github.com/eldruin/lsm303agr-rs/blob/3062525017e21730dacd5af219c4576dee004e8f/src/accel_mode_and_odr.rs#L151)
(same below)